src/regmod/utils.py

```diff
@@ -8,8 +8,32 @@
 from xspline import XSpline
 
 
-def default_vec_factory(vec: Any, size: int, default_value: float,
+def default_vec_factory(vec: Any,
+                        size: int,
+                        default_value: float,
```

Comment:
I might split this into two separate functions for 1) creating a vector and 2) validating a vector, to make the arguments clearer. For example, since `default_value` doesn't have an assigned default value, you still need to provide an argument when using the function to validate a vector, which is confusing.

Reply:
Yes! I agree! This is not a good function, and I intend to replace this function once for all. Will work on it in a separate PR :)